app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/teidata/TEIDataFragment.java

```diff
@@ -522,6 +522,7 @@ public void showTeiImage(String filePath, String defaultIcon) {
     public void showNewEventOptions(View anchor, ProgramStage stage) {
         PopupMenu popupMenu = new PopupMenu(context, anchor);
         popupMenu.inflate(R.menu.dashboard_event_creation);
+        popupMenu.getMenu().findItem(R.id.schedulenew).setVisible(!stage.hideDueDate());
```

Comment:
Nice :+1: 

I would do a couple of things:

1) Add this check in the presenter inside void onAddNewEvent, and did a tiny unit test. I know is very small but we'll avoid checking this kind of things in espresso tests.

2) I know this name comes from SDK? (This is a personal preference that wanted to share) 

```
if (account.isEnabled) {
  // ...
}
It's easier to read than
if (!account.isDisabled) {
  // ...
}
```
So perhaps we could do something like:

If (stage.hideDueDate()) popupMenu.getMenu().findItem(R.id.schedulenew).setVisible(false)
//do the else with (true)

Reply:
added unit test. @Balcan